### PR TITLE
Upgrade to Go 1.20.3

### PR DESCRIPTION
**Pull request by pkprzekwas**
**What this PR does / why we need it**:
Use docker image with go 1.20.4 version.

**Which issue(s) this PR fixes**:

xref https://github.com/kubermatic/release/issues/11

**What type of PR is this?**

/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Update golang version to 1.20.4.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
@pkprzekwas this should be updated to Go 1.20 instead of the newest patch for Go 1.19. Would it be fine for you to do that?

**Comment by pkprzekwas** [comment]
> @pkprzekwas this should be updated to Go 1.20 instead of the newest patch for Go 1.19. Would it be fine for you to do that? Otherwise, someone else(most likely me) will have to update all these images, etc again.

I was about to bring this topic on the next sig api-ui. I can take care of this.

File: modules/api/.golangci.yml [hunk 1]
```diff
@@ -77,6 +77,8 @@ issues:
   - cyclomatic complexity [0-9]+ of func `validateOpenStackCloudSpec` is high
   # gocritic
   - singleCaseSwitch # in most cases this is the beginning of a lookup table and should be kept an obvious table
+  # govet
+  - loop variable tc captured by func literal
```
Comment (pkprzekwas) [inline modules/api/.golangci.yml:81]: It's not exactly false positive, but I don't know how to deal with it. It complains about accessing outer tc while iterating over testcase in test run function. I don't know how we could refactor that. 
Screenshot 2023-04-07 at 09 30 41

**Comment by pkprzekwas** [comment]
Corresponding infra PR https://github.com/kubermatic/infra/pull/2231

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**